### PR TITLE
docs(output-targets): update dist-custom-elements docs

### PR DESCRIPTION
**Pull request by tanner-reits**
This commit rewrites and restructures portions of the dist-custom-elements output target docs. Additional config options are documented, and more context for when/how to use the target has been added.

### CODE REVIEW

File: src/docs/output-targets/custom-elements.md [hunk 1]
```diff
@@ -75,20 +68,42 @@ export const config: Config = {
...
+_default: `false`_
+
+Setting this flag to `true` will trigger inline dynamic imports for the under-the-hood Rollup bundle output. For more information, see [Rollup's documentation](https://rollupjs.org/guide/en/#outputinlinedynamicimports) for this option.
+
+### includeGlobalScripts
+
+_default: `false`_
+
+Setting this flag to `true` will include global scripts in the bundle and execute them once the bungle entry point in loaded.
+
```
Comment (tanner-reits) [inline src/docs/output-targets/custom-elements.md:106]: Omitted `minify` option here. I'll add that to my `externalRuntime` PR since it relates to some changes there

File: src/docs/output-targets/custom-elements.md [hunk 2]
```diff
@@ -192,6 +221,8 @@ your package can find the type definitions, like so:
 }
 ```
 
+> Note: If you set the `dir` property on the output target config, replace `dist/components` in the above snippet with the path set in the config.
+
 ## Example Bundler Configs
```
Comment (tanner-reits) [inline src/docs/output-targets/custom-elements.md:226]: Left this alone. I'm not too sure of it's helpfulness, but it looks like it was added ~2 years ago by a community member. Long term, I'd even consider use separating these output target docs into two pages: one for setting up/configuring, and one for consuming the output in a project.

File: src/docs/output-targets/custom-elements.md [hunk 3]
```diff
@@ -271,15 +302,7 @@ export default {
 };
 ```
 
-## How is this different from the "dist" and the "dist-custom-element-bundle" output targets?
```
Comment (tanner-reits) [inline src/docs/output-targets/custom-elements.md:274]: Killed this since bundle is going away in v3

**Review by tanner-reits — COMMENTED** [review]
This will have conflicts w/ my other two open doc PRs. I'll address those as these start to merge.

Called out a couple things, happy to add context or clarity anywhere. Also went ahead and added the missing config options. I figure if we support them, we might as well document them

File: src/docs/output-targets/custom-elements.md [hunk 4]
```diff
@@ -6,51 +6,44 @@ contributors:
   - adamdbradley
   - rwaskiewicz
   - splitinfinities
+  - tanner-reits
 ---
 
 # Custom Elements
 
-The `dist-custom-elements` output target is used to generate custom elements in a more optimized way for tree shaking, and its the recommended approach when using any frontend framework integrations. For example, if a component library has 100 components, but an external project only imported one component from the directory, then only the code used by that one component would be pulled into the project. This is due to Stencil's use of ES Modules and the compiler generating friendly code for bundlers to parse and understand.
+The `dist-custom-elements` output target is used to generate custom elements in a more optimized way for tree shaking, and it's the recommended approach when using any frontend framework integrations.
```
Comment (rwaskiewicz) [inline src/docs/output-targets/custom-elements.md:14]: The "in a more optimized way for tree shaking" was originally written (albeit unclearly) to differentiate it from `dist-custom-elements-bundle`. Since we'll be removing that in the near future, I wonder if we should doctor this sentence a little bit to either
- compare it against `www`
- talk about it a little more generally - e.g. "what's it do?"

File: src/docs/output-targets/custom-elements.md [hunk 5]
```diff
@@ -115,19 +130,42 @@ Make sure to copy the assets over to a public directory in your app. This config
 bundling, and where your assets can be loaded from. How the files are copied to the production build directory depends on the bundler or tooling.
 The configs below provide examples of how to do this automatically with popular bundlers.
 
+## Consuming Custom Elements
```
Comment (rwaskiewicz) [inline src/docs/output-targets/custom-elements.md:133]: I'd consider putting this above the "Making Assets Available". I have a feeling this will get more views than that section (but I could be wrong 🤷)

File: src/docs/output-targets/custom-elements.md (hunk 1 above)
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:106]: Yeah! I'll cherry-pick over the commit from the other PR that I want to have for this to reference

File: src/docs/output-targets/custom-elements.md [hunk 6]
```diff
@@ -75,20 +68,42 @@ export const config: Config = {
 | `auto-define-custom-elements` | A component and its children will be automatically defined with the `CustomElementRegistry` when the component's module is imported.                                                                                                                                                                                                                                                                      |
 | `single-export-module`        | All component and custom element definition helper functions will be exported from the `index.js` file in the output directory (see [Defining Exported Custom Elements](#defining-exported-custom-elements) for more information on this file's purpose). This file can be used as the root module when distributing your component library, see [below](#distributing-custom-elements) for more details. |
 
-> Note: At this time, components created not using JSX may not be automatically
-> defined. This is a known limitation of the API and users should be aware of
-> it
+> Note: At this time, components created not using JSX may not be automatically defined. This is a known limitation of the API and users should be aware of it.
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:71]: Looks like @alicewriteswrongs has a PR open to change this block

File: src/docs/output-targets/custom-elements.md [hunk 7]
```diff
@@ -6,51 +6,44 @@ contributors:
...
 
-## Config
+An array of [copy tasks](/docs/copy-tasks) to be executed during the build process.
 
 ### customElementsExportBehavior
 
-This config option provides additional behaviors that will alter the default component export OR custom element definition behaviors
+_default: `'default'`_
+
+This config option provides additional behaviors that will alter the default component export _OR_ custom element definition behaviors
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:46]: I like that 👍

File: src/docs/output-targets/custom-elements.md [hunk 8]
```diff
@@ -75,20 +68,42 @@ export const config: Config = {
...
+
+_default: `false`_
+
+Setting this flag to `true` will trigger inline dynamic imports for the under-the-hood Rollup bundle output. For more information, see [Rollup's documentation](https://rollupjs.org/guide/en/#outputinlinedynamicimports) for this option.
+
+### includeGlobalScripts
+
+_default: `false`_
+
+Setting this flag to `true` will include global scripts in the bundle and execute them once the bungle entry point in loaded.
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:105]: Oh maybe I misunderstood this option. I thought it would point to https://stenciljs.com/docs/config#globalscript

File: src/docs/output-targets/custom-elements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/docs/output-targets/custom-elements.md:71]: I just thought it sounded a bit odd!

[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply src/docs/output-targets/custom-elements.md:71]: It did. My bad 🙃

File: src/docs/output-targets/custom-elements.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply src/docs/output-targets/custom-elements.md:105]: Hmmm maybe I'm wrong? Now I'm second guessing myself 😆

[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:105]: It does refer to what I linked, just a heads up

File: src/docs/output-targets/custom-elements.md (hunk 4 above)
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md:14]: I think we kind of address comparing it to `www` with the other callout you had for lazy-loading vs. non lazy-loading (since I feel like that's the big difference most users would make a decision against). 

What else would we generally talk about? Elaborate more on treeshaking? Talk about using Rollup under-the-hood and what that generates? In that case, we might just keep what was originally there but wordsmith it a bit

File: src/docs/config/overview.md [hunk 9]
```diff
@@ -48,6 +48,9 @@ bundles: [
 ]
```
Comment (tanner-reits) [inline src/docs/config/overview.md:48]: All changes from this file are the same as from #924

File: src/docs/output-targets/custom-elements.md [hunk 10]
```diff
@@ -75,20 +68,42 @@ export const config: Config = {
...
 
+_default: `true`_
+
 By default, Stencil will generate type declaration files (`.d.ts` files) as a part of the `dist-custom-elements` output target through the `generateTypeDeclarations` field on the target options. Type declaration files will be placed in the `dist/types` directory in the root of a Stencil project. At this time, this output destination is not able to be configured.
 
 Setting this flag to `false` will not generate type declaration files for the `dist-custom-elements` output target.
 
-This flag defaults to `true` when omitted from a Stencil configuration file.
-
 > When set to generate type declarations, Stencil respects the export behavior selected via `customElementsExportBehavior` and generates type declarations specific to the content of that file.
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: Yeah this is weird cuz really what it does is generate type declarations for the `component.d.ts` file and then just re-export the types from `dist/types`. Changed this up a bit to call out what we were really trying to call attention to

File: src/docs/output-targets/custom-elements.md [hunk 11]
```diff
@@ -75,20 +68,42 @@ export const config: Config = {
...
 
+### inlineDynamicImports
+
+_default: `false`_
+
+Setting this flag to `true` will trigger inline dynamic imports for the under-the-hood Rollup bundle output. For more information, see [Rollup's documentation](https://rollupjs.org/guide/en/#outputinlinedynamicimports) for this option.
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: Think I got it in layman's terms

Also, I think there might be an issue with setting this property on the target rn... was seeing some build errors when playing around with it

File: src/docs/output-targets/custom-elements.md [hunk 12]
```diff
@@ -115,19 +130,42 @@ Make sure to copy the assets over to a public directory in your app. This config
 bundling, and where your assets can be loaded from. How the files are copied to the production build directory depends on the bundler or tooling.
 The configs below provide examples of how to do this automatically with popular bundlers.
 
+## Consuming Custom Elements
+
+By default, the custom elements files will be written to `dist/components/`. This directory can be configured using the output target's `dir` config.
+
+The generated files will each export a component class and will already have the styles bundled. However, this build does not define the custom elements or apply any polyfills. Any dependencies of your imported component will need to be loaded as well.
```
[the remark this answers is not in the document]
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: I took that out and added clarification for assets. Am a bit confused why this was here in the first place. Dependencies are bundled, are they not? And child components are included in a parent's `defineCustomElement` function.

File: src/docs/config/overview.md [hunk 13]
```diff
@@ -119,17 +125,29 @@ During production builds, the content of each generated file is hashed to repres
...
+## invisiblePrehydration
 
-## hashedFileNameLength
+*default: `true`*
 
-*default: `8`*
+When `true`, `invisiblePrehydration` will visually hide components before they are hydrated by adding an automatically injected style tag to the document's head. Setting `invisiblePrehydration` to `false` will not inject the style tag into the head, allowing you to style your web components pre-hydration. 
 
-When the `hashFileNames` config is set to `true`, and it is a production build, the `hashedFileNameLength` config is used to determine how many characters the file name's hash should be.
+> Note: Setting `invisiblePrehydration` to `false` will cause everything to be visible when your page is loaded, causing a more prominent Flash of Unstyled Content (FOUC). However, you can style your web component's fallback content to your preference.
```
Comment (rwaskiewicz) [inline src/docs/config/overview.md:134]: no action required - tiny diff b/w this and https://github.com/ionic-team/stencil-site/pull/924/files#diff-689c19470ae2ed5ac0b4ba2bb534871dd960e99ae1638ca7da6bab934a5dd022R134

"a" more prominent (emphasis on the article) is missing in #924 - I'm more making the note here for if/when we run into merge conflicts

File: src/docs/output-targets/custom-elements.md [hunk 14]
```diff
@@ -6,51 +6,50 @@ contributors:
...
+An array of [copy tasks](/docs/copy-tasks) to be executed during the build process.
 
 ### customElementsExportBehavior
 
-This config option provides additional behaviors that will alter the default component export OR custom element definition behaviors
+_default: `'default'`_
+
+By default, the `dist-custom-elements` output target generates a single file per component, and exports each of those files individually.
+
+In some cases, library authors may want to change this behavior, either to automatically define component children, provide a single file containing all component exports, etc.
```
Comment (alicewriteswrongs) [inline src/docs/output-targets/custom-elements.md:50]:
```suggestion
In some cases, library authors may want to change this behavior, for instance to automatically define component children, provide a single file containing all component exports, etc.
```

File: src/docs/output-targets/custom-elements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (alicewriteswrongs) [reply src/docs/output-targets/custom-elements.md:77]: Since this blockquote has now been changed on `main` should we merge `main` into the v3 branch and rebase this PR over that to make sure things are consistent? I think it would be easy to lose a change later in a merge commit somewhere.

File: src/docs/output-targets/custom-elements.md [hunk 15]
```diff
@@ -75,19 +74,81 @@ export const config: Config = {
...
 ### generateTypeDeclarations
 
+_default: `true`_
+
 By default, Stencil will generate type declaration files (`.d.ts` files) as a part of the `dist-custom-elements` output target through the `generateTypeDeclarations` field on the target options. Type declaration files will be placed in the `dist/types` directory in the root of a Stencil project. At this time, this output destination is not able to be configured.
 
 Setting this flag to `false` will not generate type declaration files for the `dist-custom-elements` output target.
 
-This flag defaults to `true` when omitted from a Stencil configuration file.
+> When set to generate type declarations, Stencil respects the export behavior selected via `customElementsExportBehavior` and generates type declarations specific to the content of the generated `component.d.ts` file.
```
Comment (alicewriteswrongs) [inline src/docs/output-targets/custom-elements.md:109]: is it specific to the generated `component.d.ts` or rather that the types that it will generate for `component.d.ts` are appropriate for what is exported given the `customElementsExportBehavior` ? Maybe I'm not understanding something correctly here.

File: src/docs/output-targets/custom-elements.md [hunk 16]
```diff
@@ -271,6 +325,8 @@ export default {
 };
 ```
 
+<!-- TODO: remove once custom elements bundle is officially not supported -->
```
Comment (alicewriteswrongs) [inline src/docs/output-targets/custom-elements.md:328]: should we put a JIRA ticket number in here for easy grepping later?

File: src/docs/output-targets/custom-elements.md [hunk 17]
```diff
@@ -75,19 +74,81 @@ export const config: Config = {
...
+
+_default: `'dist/components'`_
+
+This config option allows you to change the output directory where the compiled output for this output target will be written.
+
+### empty
+
+_default: `true`_
+
+If `true`, this config option will remove the contents of the [output directory](#dir) between builds.
```
Comment (alicewriteswrongs) [inline src/docs/output-targets/custom-elements.md:89]: what about phrasing this (and similar ones) more like

```suggestion
If `true`, the contents of the [output directory](#dir) will be removed between builds.
```

**Review by alicewriteswrongs — CHANGES_REQUESTED** [review]
a few questions and suggestions reading through

File: src/docs/output-targets/custom-elements.md (hunk 4 above)
Comment (rwaskiewicz) [reply src/docs/output-targets/custom-elements.md:14]: I think that currently (and before this PR) we didn't really do a good job at giving a high level description of this output target. It doesn't feel we didn't really answer "What is this?", rather we compared it to `dist-custom-elements` and said it was recommended for frontend integrations. I don't think I'm describing it well, but it feels like there's an executive summary/elevator pitch for this output target that could be added here. Does that make sense?

File: src/docs/output-targets/custom-elements.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (rwaskiewicz) [reply src/docs/output-targets/custom-elements.md:77]: Yah - I'll do that after we merge this into v3 (there're already conflicts there, and I'd like to do them in one fell swoop)

**Comment by rwaskiewicz** [comment]
@tanner-reits responded to everything I think you were looking for answers for (just https://github.com/ionic-team/stencil-site/pull/925/files#r999540607). LMK if I missed anything!

File: src/docs/output-targets/custom-elements.md (hunk 17 above)
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: Went a slightly different route, but did change a few phrases so they were consistent across the options

File: src/docs/output-targets/custom-elements.md (hunk 15 above)
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: Yeah I did make this confusing and botched it a bit. Pushed a change to hopefully make that clearer

File: src/docs/output-targets/custom-elements.md (hunk 4 above)
Comment (tanner-reits) [reply src/docs/output-targets/custom-elements.md]: Alright, added a few sentences that I morphed from a few other doc pages that were doing some of these comparisons

File: src/docs/output-targets/custom-elements.md [hunk 18]
```diff
@@ -6,51 +6,50 @@ contributors:
...
 ---
 
 # Custom Elements
 
-The `dist-custom-elements` output target is used to generate custom elements in a more optimized way for tree shaking, and its the recommended approach when using any frontend framework integrations. For example, if a component library has 100 components, but an external project only imported one component from the directory, then only the code used by that one component would be pulled into the project. This is due to Stencil's use of ES Modules and the compiler generating friendly code for bundlers to parse and understand.
+The `dist-custom-elements` output target is used to generate custom elements in a more optimized way for tree shaking, and it's the recommended approach when using any frontend framework integrations. This output target creates custom elements that directly extend `HTMLElement` and provides simple utility functions for easily defining these elements on the [Custom Element Registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry). As stated, this output target excels in use in frontend frameworks and projects that will handle bundling, lazy-loading, and custom element registration themselves.
```
Comment (rwaskiewicz) [inline src/docs/output-targets/custom-elements.md:14]: I like it - one reword suggestion to drop some of the duplication:
```suggestion
The `dist-custom-elements` output target creates custom elements that directly extend `HTMLElement` and provides simple utility functions for easily defining these elements on the [Custom Element Registry](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry). This output target excels in use in frontend frameworks and projects that will handle bundling, lazy-loading, and custom element registration themselves.
```

**Review by rwaskiewicz — CHANGES_REQUESTED** [review]

**Review by rwaskiewicz — APPROVED** [review]

**Review by alicewriteswrongs — APPROVED** [review]
I think this all looks good!
